main.rb

```diff
@@ -0,0 +1,130 @@
+# frozen_string_literal: true
+
+require 'json'
+require 'sinatra'
+
+before do
+  @app_title = 'メモアプリ'
+end
+
+helpers do
+  def h(text)
+    Rack::Utils.escape_html(text)
+  end
+end
+
+class MemoDB
+  DB_FILE_NAME = 'db.json'
+
+  def self.read_memos
+    generate_memo_lists['memo_lists']
+  end
+
+  def self.create_memo(title, content)
+    memos = generate_memo_lists
+    latest_id = memos['latest_id'] + 1
+    memo_lists = memos['memo_lists']
+
+    new_memo = { 'title' => title, 'content' => content }
+    memos['memo_lists'][latest_id] = new_memo
+    changed_memos = { latest_id:, memo_lists: }
+    save_memo_data(changed_memos)
+  end
+
+  def self.read_memo(memo_id)
+    memos = generate_memo_lists['memo_lists']
+    memo = memos[memo_id]
+    return nil if memo.nil?
+
+    { title: memo['title'], content: memo['content'] }
+  end
+
+  def self.update_memo_details(params)
+    memos = generate_memo_lists
+    latest_id = memos['latest_id']
+    memo_lists = memos['memo_lists']
+
+    memo_lists[params['id']] = { 'title' => params['title'], 'content' => params['content'] }
+    changed_memos = { latest_id:, memo_lists: }
+    save_memo_data(changed_memos)
+  end
+
+  def self.delete_memo(params)
+    memos = generate_memo_lists
+    latest_id = memos['latest_id']
+    memo_lists = memos['memo_lists']
+
+    memo_lists.delete(params['id'])
+    changed_memos = { latest_id:, memo_lists: }
+    save_memo_data(changed_memos)
+  end
+
+  def self.save_memo_data(changed_memos)
+    converted_memos = JSON.generate(changed_memos)
+
+    File.open(DB_FILE_NAME, 'w') do |file|
+      file.write(converted_memos)
```

Comment:
```suggestion
    File.open(DB_FILE_NAME, 'w') do |file|
      converted_memos = JSON.generate(changed_memos)
      file.write(converted_memos)
```

とした方が変数の寿命が短くなりそうです。